contracts/fees_collector_contract.py

```diff
@@ -188,17 +206,141 @@ def claim_rewards(self, user: Account, proxy: ProxyNetworkProvider):
+    
+    
+    def get_user_energy_for_week(self, user: str, proxy: ProxyNetworkProvider, week: int) -> dict:
+        data_fetcher = FeeCollectorContractDataFetcher(Address(self.address), proxy.url) 
+        raw_results = data_fetcher.get_data('getUserEnergyForWeek', [Address(user).get_public_key(), week])
+        print(raw_results)
+        if not raw_results:
+            return {}
+        decoder = TransactionDecoder()
+        user_energy_for_week = decoder.hex_to_number(raw_results)
+
+        return user_energy_for_week
+    
+    def get_current_week(self, proxy: ProxyNetworkProvider) -> int:
+        data_fetcher = FeeCollectorContractDataFetcher(Address(self.address), proxy.url)
+        raw_results = data_fetcher.get_data('getCurrentWeek')
+        if not raw_results:
+            return 0
+        current_week = int(raw_results)
+
+        return current_week
+    
+    def get_total_energy_for_week(self, proxy: ProxyNetworkProvider, week: int) -> int:
+        data_fetcher = FeeCollectorContractDataFetcher(Address(self.address), proxy.url)
+        raw_results = data_fetcher.get_data('getTotalEnergyForWeek', [week])
+        if not raw_results:
+            return 0
+        return int(raw_results)
+    
+    def get_total_rewards_for_week(self, proxy: ProxyNetworkProvider, week: int):
+        serializer = Serializer("@")
+
+        data_fetcher = FeeCollectorContractDataFetcher(Address(self.address), proxy.url)
+        raw_results = data_fetcher.get_data('getTotalRewardsForWeek', [week])
+        
+        if not raw_results:
+            return {}
+        
+        b = bytes.fromhex(raw_results)
+        
+        data = b
+
+        def create_esdt_token_payment() -> StructValue:
+            return StructValue([
+                Field("token_identifier", StringValue()),
+                Field("token_nonce", U64Value()),
+                Field("amount", BigUIntValue()),
+            ])
+
+        destination = ListValue(
+            item_creator=create_esdt_token_payment
+        )
+
+        s = serializer.deserialize_parts([data], [destination])
+
+        tokens = []
+
+        for i, item in enumerate(destination.items):
+            fields = cast(StructValue, item).fields
+            for field in fields:
+                fvalue: Any = field.value
+                tokens.append([field.name, fvalue.value])
+       
+        tokens_list = [tokens[i:i+3] for i in range(0, len(tokens), 3)] 
+
+        return tokens_list
+    
+    def get_all_stats(self, proxy: ProxyNetworkProvider, user: str, week: int):        
+        fees_collector_stats = {
+            "current_week": self.get_current_week(proxy),
+            "total_rewards_for_week": self.get_total_rewards_for_week(proxy, week),
+            "total_energy_for_week": self.get_total_energy_for_week(proxy, week)
+        }
+
+        return fees_collector_stats
+    
+
+    def get_tx_op(self, tx_hash: str, operation: dict, api: ApiNetworkProvider) -> dict:
```

Comment:
This should not belong here.
It would rather be somewhere in utils, like `utils.utils_tx.py` near the existing `get_tx_operations` method of the `NetworkProviders` class.